tests/test_labelarray.py

```diff
@@ -337,3 +339,37 @@ def test_setitem_array(self):
         # Write the whole array.
         arr[:] = orig_arr
         check_arrays(arr, orig_arr)
+
+    def test_narrow_code_storage(self):
+        def create_categories(width, plus_one):
+            length = int(width / 8) + plus_one
+            return [
+                ''.join(cs)
+                for cs in take(
+                    2 ** width + plus_one,
+                    product([chr(c) for c in range(256)], repeat=length),
+                )
+            ]
+
+        #  uint8
+        categories = create_categories(8, plus_one=False)
```

Comment:
The argument there is `width`, not `length`.

```
(Pdb) p len(create_categories(8, False))
256
(Pdb) p len(create_categories(8, True))
257
```
The missing value is in the categories by construction.

Also, a uint8 holds 256 values, not 8 ;)